libsoftwarecontainer/src/gateway/gateway.cpp

```diff
@@ -21,11 +21,30 @@
 
 namespace softwarecontainer {
 
+
+Gateway::Gateway(const std::string &id,
+                 std::shared_ptr<ContainerAbstractInterface> container,
+                 bool quickLaunchEnabled) :
+    m_id(id),
+    m_container(container),
+    m_quickLaunchEnabled(quickLaunchEnabled),
+    m_state(GatewayState::CREATED)
+{
+}
+
+std::string Gateway::id() const
+{
+    return m_id;
+}
+
 bool Gateway::setConfig(const json_t *config)
 {
-    if (m_state == GatewayState::ACTIVATED) {
-        log_error() << "Can not configure a gateway that is already activated: " << id();
-        throw GatewayError("Gateway already activated");
+    if (m_state == GatewayState::ACTIVATED && !m_quickLaunchEnabled) {
```

Comment:
I was thinking that it could make sense to write like this when writing new code and to change when we touch a code particle. Yet it is not a mandatory or so important thing.